asab/web/auth/service.py

```diff
@@ -161,6 +163,11 @@ def _prepare_mock_user_info(self):
 
 
 	async def initialize(self, app):
+		if self.Mode == AuthMode.ENABLED and self.DiscoveryService is None:
+			# if self.DiscoveryService is not None, wait for the Cache to get ready
+			await self._fetch_public_keys_if_needed()
+
+	async def _on_discovery_cache_ready(self, _):
```

Comment:
This is not a good idea - it will lead to "magic behaviour" where dev mode is different to production -> "It works on my computer".

Reply:
dev mode?

Reply:
the pubsub is not a good idea or self.Mode is not a good idea?